### PR TITLE
Add ConstDivCeil trait (div_ceil, checked_div_ceil)

### PR DESCRIPTION
**Pull request by kaidokert**
## Summary by Sourcery

Introduce a const-compatible ceiling-division abstraction and implement it for primitive unsigned integers and FixedUInt.

New Features:
- Add a ConstDivCeil trait providing div_ceil and checked_div_ceil for const contexts.
- Implement ConstDivCeil for primitive unsigned integer types u8, u16, u32, u64, and u128.
- Implement ConstDivCeil for FixedUInt, including a const wrapper function and usage in const contexts.

Tests:
- Add unit tests for FixedUInt ceiling division behavior, including checked and const-evaluated variants.

## Summary by CodeRabbit

* **New Features**
  * Added ceiling-division (round-up) operations with both checked and unchecked variants.
  * Ceiling-division is available for primitive unsigned integers and for fixed-size unsigned types, including const-evaluable variants usable at compile time.
* **Tests**
  * Added runtime and compile-time tests covering exact division, rounding-up, zero/edge cases, and overflow behavior.

### CODE REVIEW

File: src/const_numtrait.rs [hunk 1]
```diff
@@ -282,6 +282,23 @@ c0nst::c0nst! {
         fn checked_next_multiple_of(self, rhs: Self) -> Option<Self>;
     }
 
+    /// Const-compatible ceiling division.
+    ///
+    /// Returns the smallest integer greater than or equal to the exact quotient.
+    pub c0nst trait ConstDivCeil: Sized + [c0nst] ConstZero + [c0nst] core::ops::Div<Output = Self> + [c0nst] core::ops::Rem<Output = Self> + [c0nst] ConstOne + [c0nst] core::cmp::Eq {
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/const_numtrait.rs]: done

File: src/fixeduint/div_ceil_impl.rs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            }
+            // Use const_div which computes both quotient and remainder in one pass
+            let mut quotient = self.array;
+            let remainder = const_div(&mut quotient, &rhs.array);
+            if const_is_zero(&remainder) {
+                Some(Self { array: quotient })
+            } else {
+                Some(Self { array: quotient } + Self::one())
+            }
+        }
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/fixeduint/div_ceil_impl.rs]: fixed
